### PR TITLE
STR-157 add support for CL rpcs

### PR DESCRIPTION
**Pull request by sapinb**
A bit hacky, but trying to keep changes minimal.
Allow calls to be routed to sub groups based on method name (only `alp_` prefix for now).
Will need to look into performance later.

### CODE REVIEW

**Review by bewakes — APPROVED** [review]
Great work!

File: src/rpc/types.rs [hunk 1]
```diff
@@ -23,9 +23,34 @@ pub struct Status {
...
+}
+
+impl RouteGroup {
+    pub fn from_method_name(method_name: &str) -> Self {
+        if method_name.starts_with("alp_") {
+            Self::StrataCL
+        } else {
+            Self::Ethereum
+        }
+    }
```
Comment (delbonis) [inline src/rpc/types.rs:41]: I'd probably break this down and make it return an error if it fails to parse, so we can handle the fallback higher up the chain.  Ethereum RPC names are [pretty well-defined](https://ethereum.org/en/developers/docs/apis/json-rpc/).

File: src/rpc/types.rs [hunk 2]
```diff
@@ -23,9 +23,34 @@ pub struct Status {
...
+
+    pub fn from_config(group: Option<&str>) -> Self {
+        match group {
+            Some("strata") => Self::StrataCL,
+            _ => Self::Ethereum,
+        }
+    }
```
Comment (delbonis) [inline src/rpc/types.rs]: Just make this be a `FromStr` impl and support failing to parse.  If someone passed `"solana"` it wouldn't make sense to return `Ethereum`.

File: src/rpc/types.rs [hunk 3]
```diff
@@ -133,6 +165,11 @@ impl Rpc {
 
     /// Request blocknumber and return its value
     pub async fn block_number(&self) -> Result<u64, crate::rpc::types::RpcError> {
+        // skip sync check on strata rpc
+        if self.group == RouteGroup::StrataCL {
+            return Ok(0);
+        }
```
Comment (delbonis) [inline src/rpc/types.rs]: This is an Ethereum RPC isn't it?  It's more correct to make this be `self.group != RouteGroup::Ethereum`.

File: src/rpc/types.rs [hunk 4]
```diff
@@ -148,6 +185,11 @@ impl Rpc {
 
     /// Returns the sync status. False if we're synced and following the head.
     pub async fn syncing(&self) -> Result<bool, crate::rpc::types::RpcError> {
+        // skip sync check on strata rpc
+        if self.group == RouteGroup::StrataCL {
+            return Ok(false);
+        }
```
Comment (delbonis) [inline src/rpc/types.rs]: Same here.

**Review by delbonis — DISMISSED** [review]
Good start, should try to be a bit more neutral about how things are though.

File: src/rpc/types.rs (hunk 1 above)
Comment (sapinb) [reply src/rpc/types.rs:41]: I am thinking of ethereum rpcs as the 'default' and the others as special overrides. I'd like to keep this simple without explicitly listing all ethereum rpcs here.

**Pull request merged**